main/src/library/Result.flix

```diff
@@ -409,4 +409,29 @@ namespace Result {
         case Err(_) => Iterator.empty()
         case Ok(x)  => Iterator.singleton(x)
     }
+
+    ///
+    /// Returns `Ok(x)` if `f` was invoked without throwing an exception.
+    ///
+    /// If `f` throws a Java `RuntimeException`, `Err(e)` is returned
+    /// where `e` is the error message.
+    ///
+    pub def try(f: Unit -> a & ef): Result[a, String] & ef =
+        tryWith(f, () -> ())
+
+    ///
+    /// Returns `Ok(x)` if `f` was invoked without throwing an exception.
+    ///
+    /// If `f` throws a Java `RuntimeException` then `g` is invoked
+    /// and `Err(e)` is returned where `e` is the error message.
+    ///
+    pub def tryWith(f: Unit -> a & ef1, g: Unit -> b & ef2): Result[a, String] & (ef1 and ef2) =
+        try {
+            Ok(f())
+        } catch {
+            case e: ##java.lang.RuntimeException =>
```

Comment:
Flix errors are subclasses of RuntimeException (which makes sense) so they will also be caught here. I don't think this can be avoided very easily I just want to state it so you are aware. I'm not 100p sure but my understanding was that Flix programmers shouldn't write code using exceptions (no catching flix exceptions at least). We could maybe look at enforcing this at some point? One idea could be to insert `case e: FlixError => throw e` at the top of all try blocks to avoid capture but im just thinking out loud

Reply:
Does FlixError extend java.lang.Error or java.lang.RuntimeException? I think a simple solution might be to extend Error!

Reply:
Runtime at the moment. I think actually this code should catch `Exception` to catch [these](https://docs.oracle.com/javase/7/docs/api/java/io/IOException.html) fx. but maybe its hard to predict all uses.

Either way, user made code might still catch flix errors so I think we could discuss circumventions

Reply:
I think:

- We should catch Exception.
- FlixError should extend `java.lang.Error` or whatever it is called.